src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ExecutionEngineCode/BatchParserSqlCmd.cs

```diff
@@ -105,17 +105,78 @@ public override void SetVariable(PositionStruct pos, string name, string value)
             get { return internalVariables; }
             set { internalVariables = value; }
         }
-        
-        
+
+
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "ppIBatchSource")]
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "fileName")]
         public override BatchParserAction Include(TextBlock filename, out TextReader stream, out string newFilename)
         {
             stream = null;
             newFilename = null;
+            LineInfo lineInfo;
+
+            if (filename == null)
+            {
+                stream = null;
+                return BatchParserAction.Abort;
+            }
+
+            filename.GetText(resolveVariables: true, text: out newFilename, lineInfo: out lineInfo);
+            string resolvedFileNameWithFullPath = GetFilePath(newFilename);
+
+            if (!File.Exists(resolvedFileNameWithFullPath))
+            {
+                stream = null;
+                return BatchParserAction.Abort;
```

Comment:
Should we RaiseScriptError before we abort?

Reply:
Validated that the error is raised in caller depending on abort result - if we raise here end user will get two errors